src/Lykke.AlgoStore.CSharp.AlgoTemplate.Models/Repositories/StatisticsRepository.cs

```diff
@@ -13,39 +14,82 @@ namespace Lykke.AlgoStore.CSharp.AlgoTemplate.Models.Repositories
     public class StatisticsRepository : IStatisticsRepository
     {
         private readonly INoSQLTableStorage<StatisticsEntity> _table;
+        private readonly INoSQLTableStorage<StatisticsSummaryEntity> _tableSummary;
 
-        public static readonly string TableName = "Statistics";
+        public static readonly string TableName = "AlgoStatistics";
 
-        public StatisticsRepository(INoSQLTableStorage<StatisticsEntity> table)
+        public StatisticsRepository(
+            INoSQLTableStorage<StatisticsEntity> table,
+            INoSQLTableStorage<StatisticsSummaryEntity> tableSummary)
         {
             _table = table;
+            _tableSummary = tableSummary;
         }
 
-        public static string GeneratePartitionKey(string key) => key;
+        public static string GeneratePartitionKey(string instanceId, AlgoInstanceType instanceType) => $"{instanceId}_{instanceType}";
 
         public static string GenerateRowKey(string key) => String.IsNullOrEmpty(key) ? DateTime.UtcNow.ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fff'Z'") : key;
 
+        public static string GenerateSummaryRowKey() => "Summary";
+
         public async Task CreateAsync(Statistics data)
         {
             var entity = AutoMapper.Mapper.Map<StatisticsEntity>(data);
-
-            entity.PartitionKey = GeneratePartitionKey(data.InstanceId);
+            entity.PartitionKey = GeneratePartitionKey(data.InstanceId, data.InstanceType);
             entity.RowKey = GenerateRowKey(data.Id);
 
-            await _table.InsertAsync(entity);
+            var entitySummary = await _tableSummary.GetDataAsync(
+                GeneratePartitionKey(data.InstanceId, data.InstanceType),
+                GenerateSummaryRowKey());
+
+            if (entitySummary == null)
+                throw new ArgumentException("Statistics summary row not found");
+
+            if (data.IsStarted.HasValue && data.IsStarted.Value)
+                entitySummary.TotalNumberOfStarts++;
+
+            if (data.IsBuy.HasValue)
+            {
+                entitySummary.TotalNumberOfTrades++;
+
+                if(!entity.Amount.HasValue)
+                    throw new ArgumentException("Amount for statistics not provided");
+
+                if (!entity.Price.HasValue)
+                    throw new ArgumentException("Price for statistics not provided");
+
+                if (data.IsBuy.Value)
+                {
```

Comment:
This should be different depending on which asset we trade. I also think this values should be calculated before we call the repo. When we have the life version we should use the trading service and then extract the real values of assets after trading.